URITemplate/URITemplate.swift

```diff
@@ -259,7 +259,8 @@ extension NSRegularExpression {
 
 extension String {
   func percentEncoded() -> String {
-    return CFURLCreateStringByAddingPercentEscapes(nil, self, nil, ":/?&=;+!@#$()',*", CFStringConvertNSStringEncodingToEncoding(NSUTF8StringEncoding)) as String
+    let allowedCharacters = NSCharacterSet(charactersInString: ":/?&=;+!@#$()',*").invertedSet
+    return stringByAddingPercentEncodingWithAllowedCharacters(allowedCharacters)!
```

Comment:
I only did this because I saw that things were being force unwrapped in similar methods.

Happy to make the return type optional and try to address the repercussions if necessary.